client_sdk/go/fpc/crypto/crypto.go

```diff
@@ -29,16 +30,65 @@ import "C"
 
 var logger = flogging.MustGetLogger("fpc-client-crypto")
 
-func encrypt(input []byte, encryptionKey []byte) ([]byte, error) {
-	return input, nil
+func keyGen() ([]byte, error) {
+	// generate random symmetric key of 16 bytes
+	// the length is required by the pdo crypto library
+	key := make([]byte, 16)
+	n, err := rand.Read(key)
+	if n != len(key) || err != nil {
+		return nil, err
+	}
+
+	return key, nil
 }
 
-func keyGen() ([]byte, error) {
-	return []byte("fake key"), nil
+func encrypt(input []byte, encryptionKey []byte) ([]byte, error) {
+	//This is an RSA encryption performed with the pdo crypto library
+	//Importantly, the library uses 2048bit RSA keys, so the input size has to be ~200bytes
+	//TODO-1: bump up the key length to 3072
+	//TODO-2: extend procedure for large input sizes
+
```

Comment:
Some additional background/motivation
```suggestion
	//Importantly, the library uses 2048bit RSA keys & OAEP encoding, so the input size can be at most ~200bytes
	//TODO-1: bump up the key length to 3072 to match NIST strength
	//TODO-2: extend procedure for large input sizes (via hybrid encryption)

```

Reply:
copy-pasted.